### PR TITLE
OSM Tag Schemes

### PR DESCRIPTION
**Pull request by droogmic**
Apologies, this PR suffered from scope creep :/

### CODE REVIEW

File: osm-tag-schemes/src/highway.rs [hunk 1]
```diff
@@ -228,6 +208,24 @@ impl Highway {
         }
     }
 
+    /// Active Highway
```
Comment (droogmic) [inline osm-tag-schemes/src/highway.rs:211]: construction

File: osm-tag-schemes/src/highway.rs [hunk 2]
```diff
@@ -228,6 +208,24 @@ impl Highway {
...
+    #[must_use]
+    pub fn construction(r#type: HighwayType) -> Self {
+        Self {
+            highway: r#type,
+            lifecycle: Lifecycle::Construction,
+        }
+    }
+
+    /// Active Highway
```
Comment (droogmic) [inline osm-tag-schemes/src/highway.rs:220]: proposed

File: osm2lanes/src/tag_keys.rs [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (droogmic) [inline osm2lanes/src/tag_keys.rs:1]: delete

**Comment by BudgieInWA** [comment]
Sometimes a band-aid needs to be ripped off, and the changes need to be made together :P 

This looks exciting! What are your goals here?

**Comment by droogmic** [comment]
This extends your design of having a semantic tagging scheme exist as an intermediate step, that can be useful for more than just our client.

If `Tags` exists in a basic crate, then we build another crate on top of it that does tagging scheme parsing, and then we put osm2lanes on top of both :)

**Comment by BudgieInWA** [comment]
Great! A description of this setup in the README (under structure?) would be awesome.

I agree that the description of individual tagging schemes are complicated enough on their own, and should exist independently like that. It feels like a place to collect together the documentation for what all the variants mean, how they should be interpreted, (links to examples?), references to the wiki and usage, ...

It is like they ought to extend / merge with the wiki pages [(e.g.)](https://wiki.openstreetmap.org/wiki/Tag:cycleway=lane).

**Pull request merged**